rootfs/scheduler/__init__.py

```diff
@@ -568,42 +583,53 @@ def run(self, namespace, name, image, entrypoint, command, **kwargs):
         if unhealthy(response.status_code):
             error(response, 'create Pod in Namespace "{}"', namespace)
 
-        data = ''
-        duration = 30
-        iteration = 1
-        while (iteration < duration):
-            try:
+        labels = {
+            'app': namespace,
+            'type': 'run',
+            'version': kwargs.get('version'),
+            'heritage': 'deis',
+        }
+        # wait for run pod to start - use the same function as scale
+        self._wait_until_pods_are_ready(namespace, container, labels, desired=1)
+
+        try:
+            # give pod 20 minutes to execute (after it got into ready state)
+            # this is a fairly arbitrary limit but the gunicorn worker / LBs
+            # will make this timeout around 20 anyway.
+            # TODO: Revisit in the future so it can run longer
+            state = 'up'  # pod is still running
+            waited = 0
+            timeout = 1200  # 20 minutes
```

Comment:
We can't do that since that's the overall timeout for the worker and there are many other pod waits ahead of this `1200` second limit. If we used the gunicorn timeout then the worker would always timeout before the `while` loop would catch - I felt it was better to have a hardcoded one, not based on the gunicorn worker limit.